src/com/ms/silverking/numeric/NumConversion.java

```diff
@@ -5,364 +5,345 @@
+  }
+
+  public static void intToBytesLittleEndian(int value, byte[] b, int offset) {
+    b[offset + 3] = (byte) (value >>> 24);
+    b[offset + 2] = (byte) (value >> 16 & 0xff);
+    b[offset + 1] = (byte) (value >> 8 & 0xff);
+    b[offset + 0] = (byte) (value & 0xff);
+  }
+
+  public static void intToBytes(int value, byte[] b, int offset) {
+    b[offset + 0] = (byte) (value >>> 24);
+    b[offset + 1] = (byte) (value >> 16 & 0xff);
+    b[offset + 2] = (byte) (value >> 8 & 0xff);
+    b[offset + 3] = (byte) (value & 0xff);
+  }
+
+  public static void intToBytes(int value, byte[] b) {
+    intToBytes(value, b, 0);
+  }
+
+  public static byte[] intToBytes(int value) {
+    byte[] b;
+
+    b = new byte[BYTES_PER_INT];
+    intToBytes(value, b);
+    return b;
+  }
+
+  /**
+   * Return a byte to an integer in the range [0, 255]
+   *
+   * @return
+   */
+  public static int byteToPositiveInt(byte b) {
+    return ((int) b) & 0xff;
+  }
+
+  public static int unsignedByteToInt(byte b) {
+    return ((int) b) & 0xff;
+  }
+
+  public static int unsignedByteToInt(byte[] b, int offset) {
+    return unsignedByteToInt(b[offset]);
+  }
+
+  ////////
+  // long
+
+  public static long bytesToLong(byte b0, byte b1, byte b2, byte b3, byte b4, byte b5, byte b6, byte b7) {
+    long value;
+
+    value =
+        (((long) b0 << 56) & 0xff00000000000000L) | (((long) b1 << 48) & 0x00ff000000000000L) | (((long) b2 << 40) & 0x0000ff0000000000L) | (((long) b3 << 32) & 0x000000ff00000000L) | (((long) b4 << 24) & 0x00000000ff000000L) | (((long) b5 << 16) & 0x0000000000ff0000L) | (((long) b6 << 8) & 0x000000000000ff00L) | (((long) b7) & 0x00000000000000ffL);
+    return value;
+  }
+
+  public static long bytesToLong(byte[] b) {
+    return bytesToLong(b[0], b[1], b[2], b[3], b[4], b[5], b[6], b[7]);
+  }
+
+  public static long bytesToLong2(byte[] b) {
+    return bytesToLong(b, 0);
+  }
+
+  public static long bytesToLong(byte[] b, int offset) {
+    return bytesToLong(b[offset + 0], b[offset + 1], b[offset + 2], b[offset + 3], b[offset + 4], b[offset + 5],
+        b[offset + 6], b[offset + 7]);
+  }
+
+  public static long bytesToLongLittleEndian(byte[] b) {
+    return bytesToLong(b[7], b[6], b[5], b[4], b[3], b[2], b[1], b[0]);
+  }
+
+  public static long bytesToLongLittleEndian(byte[] b, int offset) {
+    return bytesToLong(b[offset + 7], b[offset + 6], b[offset + 5], b[offset + 4], b[offset + 3], b[offset + 2],
+        b[offset + 1], b[offset + 0]);
+  }
+
+  public static void longToBytes(long value, byte[] b, int offset) {
+    b[offset + 0] = (byte) (value >> 56 & 0xff);
+    b[offset + 1] = (byte) (value >> 48 & 0xff);
+    b[offset + 2] = (byte) (value >> 40 & 0xff);
+    b[offset + 3] = (byte) (value >> 32 & 0xff);
+    b[offset + 4] = (byte) (value >> 24 & 0xff);
+    b[offset + 5] = (byte) (value >> 16 & 0xff);
+    b[offset + 6] = (byte) (value >> 8 & 0xff);
+    b[offset + 7] = (byte) (value & 0xff);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 224:231, 235:242.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.